contracts/staking/contracts/src/staking_pools/MixinStakingPoolRewardVault.sol

```diff
@@ -32,6 +33,38 @@ contract MixinStakingPoolRewardVault is
     MixinStorage
 {
 
+    /// @dev Asserts that the sender is the operator of the input pool.
+    /// @param poolId Pool sender must be operator of.
+    modifier onlyStakingPoolOperator(bytes32 poolId) {
+        address poolOperator = rewardVault.operatorOf(poolId);
+        if (msg.sender != poolOperator) {
+            LibRichErrors.rrevert(LibStakingRichErrors.OnlyCallableByPoolOperatorError(
+                msg.sender,
+                poolOperator
+            ));
+        }
+
+        _;
+    }
+
+    /// @dev Asserts that the sender is the operator of the input pool or the input maker.
+    /// @param poolId Pool sender must be operator of.
+    /// @param makerAddress Address of a maker in the pool.
+    modifier onlyStakingPoolOperatorOrMaker(bytes32 poolId, address makerAddress) {
+        address poolOperator = rewardVault.operatorOf(poolId);
+        if (msg.sender != poolOperator && msg.sender != makerAddress) {
```

Comment:
We can make the case where `msg.sender != makerAddress` cheaper by only calling the function `operatorOf` when we are sure that `msg.sender == makerAddress`. Solidity implements short-circuiting in high-level `if-statements`, so I think that one really clean way to refactor this would be to write:

```
address poolOperator;
if (
    msg.sender != makerAddress &&
    msg.sender != (poolOperator = rewardVault.operatorOf(poolId))
) { ... }
```

I've tested something like this, and it definitely works, so feel free to rip that off. I believe that this is as efficient as this will get, but I'd love to see any alternatives that you come up with :)